fix double slashed paths in DatabaseOrdinary by switching to std::filesystem::path

bharatnc opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement


Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Improve path concatenation and fix double slashed paths using std::filesystem::path instead of std::string in `DatabaseOrdinary.cpp`.

Detailed description / Documentation draft:

The concatenation of path doesn't always happen reliably and can cause outputs which contain double slashed paths: (Referencing from issue #8301 and this PR relates to that)
```bash
2019.12.19 17:33:18.401719 [ 1 ] {} <Error> Application: DB::Exception: Cannot parse definition from metadata file /var/lib/clickhouse/metadata/default//aaaa.sql, error: DB::Exception: Syntax error (in file /var/lib/clickhouse/metadata/default//aaaa.sql): failed at position 21 (line 3, col 1): )
```
To fix this I have chosen `std::filesystem::path` to concatenate the file paths instead of using `std::string`.

I have also removed several imports that are not being used. Also `clang` formatted the file and it looks like that did generate several changes to formatting in `DatabaseOrdinary.cpp`.

bharatnc committed:
remove unused imports

bharatnc committed:
clang format file

bharatnc committed:
construct path using boost::filesystem::path

bharatnc committed:
fix style check - clang format for lambdas

bharatnc commented on src/Databases/DatabaseOrdinary.cpp:
```diff
@@ -200,10 +207,14 @@ void DatabaseOrdinary::startupTables(ThreadPool & thread_pool)
     std::atomic<size_t> tables_processed{0};
 
-    auto startup_one_table = [&](const StoragePtr & table) {
+    // clang-format off
```
Looks like clang-format doesn't respect brace wrapping for lambdas. Looks like it passes the style checks now:
https://clickhouse-test-reports.s3.yandex.net/11900/1244ca562759c4b92a08f4350080cc01567e8c99/style_check.html

Before it failed: https://clickhouse-test-reports.s3.yandex.net/11900/ea40e7b4039f69e8e1cb20ce23039ed0c93bc7ca/style_check.html#fail1

bharatnc committed:
remove clang-format tag comments

tavplubix commented:
The changes LGTM, but could you please replace `boost::filesystem::path` with `std::filesystem::path`? You're right, some parts of boost are widely used in ClickHouse codebase, but there is no usages of `boost::filesystem` and we prefer `std::filesystem::path` or `Poco::Path` (probably will be also replaced with `std::filesystem` sometime).

bharatnc committed:
use std::filesystem::path

bharatnc commented:
> The changes LGTM, but could you please replace `boost::filesystem::path` with `std::filesystem::path`? You're right, some parts of boost are widely used in ClickHouse codebase, but there is no usages of `boost::filesystem` and we prefer `std::filesystem::path` or `Poco::Path` (probably will be also replaced with `std::filesystem` sometime).

Sure 👍. I've changed to std:: filesystem::path.
